editoast/src/views/infra/mod.rs

```diff
@@ -141,6 +146,51 @@ async fn create(
     .await
 }
 
+/// Duplicate an infra
+#[post("/<infra>/clone")]
+async fn clone(infra: i64, conn: DBConnection) -> ApiResult<Custom<Json<i64>>> {
+    let mut model_table_names = vec![];
+    let mut layer_table_names = vec![];
```

Comment:
Building these vec is not needed. We can iterate over the tables and duplicate the objects.